WeakAuras/Prototypes.lua

```diff
@@ -11244,9 +11288,150 @@ Private.event_prototypes = {
+          if active and tristateValue ~= "nil" and condition then
+            if tostring(attribute) ~= tristateValue then
+              active = false
+              primaryCheckFailed = primary and true
+            end
+          end
+        end
+
+        -- Check tristate values
+        checkTristate(discovered, currencyInfo.discovered, true, true)
+        checkTristate(capped, currencyInfo.quantity >= (currencyInfo.maxQuantity or 0), currencyInfo.maxQuantity and currencyInfo.maxQuantity > 0, false)
+        checkTristate(seasonCapped, currencyInfo.totalEarned >= (currencyInfo.maxQuantity or 0), currencyInfo.useTotalEarnedForMaxQty and currencyInfo.maxQuantity and currencyInfo.maxQuantity > 0, true)
+        checkTristate(weeklyCapped, currencyInfo.quantityEarnedThisWeek >= (currencyInfo.maxWeeklyQuantity or 0), currencyInfo.maxWeeklyQuantity and currencyInfo.maxWeeklyQuantity > 0, true)
+
+        states[""] = states[""] or {}
+        local state = states[""]
+        state.show = active
+        state.changed = true
+        state.name = currencyInfo.name
+        state.icon = currencyInfo.iconFileID
+        state.value = currencyInfo.quantity
+        state.total = currencyInfo.maxQuantity
+        state.progressType = "static"
+        state.accountQuantity = currencyInfo.accountQuantity
+        state.realAccountQuantity = currencyInfo.realAccountQuantity
+        state.realCharacterQuantity = currencyInfo.realCharacterQuantity
+        state.maxQuantity = currencyInfo.maxQuantity
+        state.quantityEarnedThisWeek = currencyInfo.quantityEarnedThisWeek
+        state.totalEarned = currencyInfo.totalEarned
+        state.capped = currencyInfo.maxQuantity and currencyInfo.maxQuantity > 0 and currencyInfo.quantity >= currencyInfo.maxQuantity
+        state.seasonCapped = currencyInfo.useTotalEarnedForMaxQty and currencyInfo.maxQuantity and currencyInfo.maxQuantity > 0 and currencyInfo.totalEarned >= currencyInfo.maxQuantity
+        state.weeklyCapped = currencyInfo.maxWeeklyQuantity and currencyInfo.maxWeeklyQuantity > 0 and currencyInfo.quantityEarnedThisweek >= currencyInfo.maxWeeklyQuantity
+        state.description = currencyInfo.description
+        state.quality = currencyInfo.quality
+        state.discovered = currencyInfo.discovered
+        state.transferPercentage = currencyInfo.transferPercentage
+        state.unitName = UnitName("player")
+        state.unitGUID = UnitGUID("player")
+        state.mainCharacter = true
+
+        if clone and not primaryCheckFailed then
+          for i, currencyData in ipairs(accountCurrencyData) do
+            local cloneId = currencyData.characterGUID or tostring(i)
+            states[cloneId] = CopyTable(states[""])
+            local s = states[cloneId]
+            s.value = currencyData.quantity
+            s.realCharacterQuantity = currencyData.realQuantity
+            s.unitName = currencyData.characterName
+            s.unitGUID = currencyData.characterGUID
```

Comment:
there is no `unit` involved in this trigger, so i would rename these to `characterName` and `GUID`